presto-hive-metastore/src/main/java/com/facebook/presto/hive/PartitionExtraInfoFetcher.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import org.apache.hadoop.hive.metastore.api.Partition;
+
+import java.util.Optional;
+
+public interface PartitionExtraInfoFetcher
```

Comment:
Should we just name this `PartitionVersionFetcher`?

Reply:
I kept the name generic so that we can add more methods to it in future. Like methods that may combine multiple values from Partition parameters and return. WDYT ?

Reply:
Maybe we should keep it simple at the beginning and change it once it's needed. Being generic could also mean being a little big vague :p

Reply:
Haha no issues. i will rename it to PartitionVersionFetcher.